### PR TITLE
feat: implement saved contact address list item component

### LINKED ISSUE #17387: Implement List Items > Saved Contact Address component
# Feature

Implement the "Saved Contact Address" component from the List Items category in the quo2 component library.

### Designs

[Link to Figma](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=21655-211666&mode=design&t=HCH4qdKJOzRT9hd2-4)

### Acceptance Criteria:

- Should follow the [best practices](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md)
- Should add [component test](https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md)
- Should add an export from the `quo2.core` namespace
- Add a preview for this component under the `list-items` category in the Quo2 preview screen

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #17387  

### Summary

Implement Saved Contact Address List Item component

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Open Quo2 previews
- Select List Items > Saved Contact Address
- Verify correct design and behavior of the component

status: ready

### CODE REVIEW

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            [quo2.theme :as quo.theme]
+            [react-native.core :as rn]
+            [quo2.components.list-items.saved-contact-address.style :as style]
+            [reagent.core :as reagent]
+            [quo2.components.icon :as icon]
+            [utils.i18n :as i18n]))
+
+(defn- f-internal-view
+  []
+  (let [state       (reagent/atom :default)
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/saved_contact_address/view.cljs:36]: I think we need to expose the `state` prop to control the component from the parent. Maybe, just for the `active` state. WYDT?

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                               (reset! active? (= new-state :active))
+                               (reset! state new-state)
+                               (when on-press
+                                 (on-press))))]
+        [rn/pressable
+         {:style               (style/container
+                                {:state @state :customization-color customization-color})
+          :on-press-in         on-press-in
+          :on-press-out        on-press-out
+          :accessibility-label :container}
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/saved_contact_address/view.cljs:62]: I believe we will need an `on-press` prop.

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            [rn/view
+             {:style               style/contact-icon-container
+              :accessibility-label :contact-icon}
+             [icon/icon :i/contact
+              {:no-color true
+               :size     12
+               :color    (colors/theme-colors
+                          (colors/custom-color :blue 50)
+                          (colors/custom-color :blue 60)
+                          theme)}]]]
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/saved_contact_address/view.cljs]: The `color` won't be applied if we pass true to the `no-color` prop.

P.S. If the icon has two colours, we need to use SVG. (ref `quo2.components.icons.svg` ns)

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+               {:size   :paragraph-2
+                :weight :monospace
+                :style  (style/account-name theme)}
+               (:name account-props)]
+              [rn/view {:style (style/dot-divider theme)}]
+              [text/text
+               {:size   :paragraph-2
+                :weight :monospace
+                :style  (style/account-address theme)}
+               (:address account-props)]]
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/saved_contact_address/view.cljs]: It's good to have the `account-props` de-structured for better readability and maintainability.

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+              {:accessibility-label :accounts-count
+               :size                :paragraph-2
+               :style               (style/accounts-count theme)}
+              (i18n/label :t/accounts-count {:count accounts-count})])]]
+         [rn/view {:accessibility-label :check-icon}
+          [icon/icon :i/chevron-right
+           {:size  20
+            :color (colors/theme-colors colors/neutral-50
+                                        colors/neutral-40
+                                        theme)}]]]))))
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/saved_contact_address/view.cljs]: - I believe we will show the `chevron-right` icon if we have more than one account.
 - If the wrapper view is used only for accessibility, we can remove it and pass the `accessibility-label` to the icon.

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+(defn- internal-view
+  [props]
+  [:f> f-internal-view props])
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/saved_contact_address/view.cljs]: @briansztamfater - any reason why we use a functional component here?

File: src/quo2/components/list_items/saved_contact_address/style.cljs [hunk 7]
```diff
@@ -0,0 +1,61 @@
+(ns quo2.components.list-items.saved-contact-address.style
+  (:require [quo2.foundations.colors :as colors]))
+
+(defn- background-color
+  [{:keys [state customization-color]}]
+  (cond (or (= state :pressed) (= state :selected))
```
Comment (J-Son89) [inline src/quo2/components/list_items/saved_contact_address/style.cljs:6]: should this not be themed? 🤔

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+              :size   :paragraph-1}
+             (:full-name contact-props)]
+            [rn/view
+             {:style               style/contact-icon-container
+              :accessibility-label :contact-icon}
+             [icon/icon :i/contact
+              {:no-color true
+               :size     12
+               :color    (colors/theme-colors
+                          (colors/custom-color :blue 50)
```
Comment (J-Son89) [inline src/quo2/components/list_items/saved_contact_address/view.cljs]: surely this is also not hardcoded as `:blue` right? and it needs to be `customization-color` instead?

File: src/quo2/components/list_items/saved_contact_address/component_spec.cljs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+(def account
+  {:name                "New House"
+   :address             "0x21a...49e"
+   :emoji               "🍔"
+   :customization-color :purple})
+
+(h/describe "List items: saved contact address"
+  (h/test "default render"
+    (h/render [saved-contact-address/view {:accounts (repeat 1 account)}])
```
Comment (vkjr) [inline src/quo2/components/list_items/saved_contact_address/component_spec.cljs]: I'd suggest having one test rendering component without any props. 
Some components tend to crash on this and that makes dev life harder when you try to use them later without knowing correct props.

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 3 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs]: True! Thanks for the suggestion, added the icon as SVG as I need to slightly change the backgrond color while preserving the white shapes inside

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 2 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs:62]: I added it, not too much change in the behavior from calling it in `on-press-out`, but it feels more natural by calling it in on-press 👍

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 5 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs]: True! Nice catch, fixed it

File: src/quo2/components/list_items/saved_contact_address/component_spec.cljs (hunk 9 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/component_spec.cljs]: Yeah, this test it supposed to have no props, nice catch! Fixed it

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 8 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs]: I changed it to use primary colors, it has some minor variation in the icon background, not supposed to be customizable

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 6 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs]: No, probably some copy-paste leftover, removed it

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs:36]: Hmm I think we would not want to expose the whole state atom, but I could add a boolean prop to set if we want to set active state? WDYT?

File: src/quo2/components/list_items/saved_contact_address/style.cljs (hunk 7 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/style.cljs:6]: Hmm its just the customization color with different opacities with not difference on the themes, so I think not

File: src/quo2/components/list_items/saved_contact_address/view.cljs [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+             (:full-name contact-props)]
+            [rn/view
+             {:style               style/contact-icon-container
+              :accessibility-label :contact-icon}
+             [icon/icon :i/contact
+              {:size    12
+               :color   (colors/theme-colors colors/primary-50 colors/primary-60 theme)
+               :color-2 colors/white}]]]
+           (if account-props
+             [account account-props theme]
```
Comment (J-Son89) [inline src/quo2/components/list_items/saved_contact_address/view.cljs]: Maybe it's cleaner to assoc the map here instead of theme as it's own param

**Review by J-Son89 — APPROVED** [review]
Nice one @briansztamfater!

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 10 above)
Comment (briansztamfater) [reply src/quo2/components/list_items/saved_contact_address/view.cljs]: Sure, updated with assoc

**Comment by briansztamfater** [comment]
@Francesca-G please review this component 🙏

**Review by Francesca-G — APPROVED** [review]
Here's the [design review ](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3374%3A300992&mode=design&t=M1KtotHHcOFvU96f-1)

Adding follow up required for minor fixes :)

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 2 above)
Comment (smohamedjavid) [reply src/quo2/components/list_items/saved_contact_address/view.cljs:62]: I agree, Brian, but we need to rely on the `on-press-out` for the animations.

I thought we would need an `on-press` for handling any events (probably navigation) on press of the component.

File: src/quo2/components/list_items/saved_contact_address/view.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/quo2/components/list_items/saved_contact_address/view.cljs:36]: Oh, no. Not the atom. I meant the `state` prop (Figma 1-1).

Most likely we might need a prop for the `selected` state. Right now, the component handles it but we need to set it from the screen which uses this component.

**Review by smohamedjavid — APPROVED** [review]
Nice work! @briansztamfater 🚀

**Comment by briansztamfater** [comment]
> Here's the [design review ](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3374%3A300992&mode=design&t=M1KtotHHcOFvU96f-1)
> 
> Adding follow up required for minor fixes :)

Thanks for testing @Francesca-G, I've adjusted the icon centering a bit, but the identicon stuff inolves implementing functionality in User Avatar component (and probably the backend) so I will create another issue to track that

**Comment by briansztamfater** [comment]
BTW @Francesca-G I'll merge this one, feel free to review the latest build, I'll create a new PR if you find other issues 👍

**Pull request merged**